File: C++/sort_vowels_in_a_string.cpp
// no.2785
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    string vowels;
    Solution(){
        vowels = "AEIOUaeiou";
    }
    bool checkVowel(char s){
        for(int i = 0; i < vowels.length(); i++){
            if(s == vowels[i]){
                return true;
            }
        }
        return false;
    }
    string sortVowels(string s) {
        vector<int> vowel_pos;
        vector<char> vs;
        for(int i = 0; i < s.length(); i++){
            if (checkVowel(s[i])){
                vs.push_back(s[i]);
                vowel_pos.push_back(i);
            }
        }
        if(vowel_pos.size()){
            sort(vs.begin(), vs.end());
            for (int i = 0; i < vowel_pos.size(); i++){
                s[vowel_pos[i]] = vs[i];
            }
        }
        return s;
    }
};

File: go/sort_vowels_in_a_string.go
//no. 2785
import(
  "sort"
  "string"
)


func checkVowel(s byte) bool{
    vowels := "AEIOUaeiou"
    for i := 0; i < len(vowels); i++{
        if s == vowels[i]{
            return true
        }
    }
    return false
}
func sortVowels(s string) string {
    var vowel_pos []int
    var vs []byte
    var chars []byte
    for i := 0; i < len(s); i++{
        chars = append(chars, s[i])
        if checkVowel(s[i]){
            vs = append(vs, s[i])
            vowel_pos = append(vowel_pos, i)
        }
    }
    if (len(vowel_pos) > 0){
        sort.Slice(vs, func(i, j int) bool {
            return vs[i] < vs[j]
        })
        for i := 0; i < len(vowel_pos); i++{
            chars[vowel_pos[i]] = vs[i]
        }
        s = string(chars[:])
    }
    return s
}
